### PR TITLE
feat(github/workflows): nixify prepare scripts, add separate PR workflow that tests unbumped versions

### PR DESCRIPTION
**Pull request by steveej**
* flake: add repo-git input

* nix/modules/scripts: add readme generation

  this script is migrated from the release-prepare github workflow.

* nix/modules/scripts: add release bump and check

  this script is migrated from the release-prepare github workflow.

* nix/modules/release-automation: split tests into unit and repo

  the repo test uses the aforementioned script to perform the same
  operations as in the release-prepare job. to access the git
  information it relies on the `.git` directory being passed in via the
  `repo-git` flake input.

* github/worfklows/release: don't run on PRs

  from now on this workflow is exclusive to release (dry) runs

* github/worfklows/pr_tests: init

  this worfklow calls the holochain-build-and-test workflow on every PR

* github/worfklows/holochain-build-and-test: add the release repo test

* github/worfklows/release-prepare: use nix scripts

### TODO:
- [x] requires #2720 

- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by jost-s — COMMENTED** [review]
This is beyond my understanding. The readme script part looks good to me.

File: flake.nix [hunk 1]
```diff
@@ -7,6 +7,10 @@
     empty.url = "github:steveej/empty";
     empty.flake = false;
 
+    # workaround to getting passing the `.git` directory into the release-automation tests
```
Comment (ThetaSinner) [inline flake.nix:10]: getting passing?

**Review by ThetaSinner — APPROVED** [review]

File: .github/workflows/holochain-build-and-test.yml [hunk 2]
```diff
@@ -41,10 +46,17 @@ jobs:
           - cmd:
               pkgs:
                 - build-holochain-build-crates-standalone
-                - build-release-automation-tests
-                - build-release-automation-tests-repo
                 - build-holochain-tests-static-all
-              extra_arg: "--override-input versions ./versions/0_2 --override-input holochain ${{ inputs.repo_path }}"
+                - build-release-automation-tests-unit
```
Comment (ThetaSinner) [inline .github/workflows/holochain-build-and-test.yml:50]: I've also made changes to this matrix as part of the cachix work, one of us will have to rebase here

File: flake.nix (hunk 1 above)
Comment (steveej) [reply flake.nix:10]:
```suggestion
    # workaround to allow the passing in of the `.git` directory into the release-automation tests
```

is this understandable?

Comment (ThetaSinner) [reply flake.nix:10]: Yes, this is good

**Comment by steveej** [comment]
- [ ] requires #2720

**Pull request merged**